src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -31,7 +39,69 @@ private void refreshTestData(int numStudents, int numQuestions) {
 
     private void refreshTestData(String filename) {
         testData = loadDataBundle(DATA_FOLDER_PATH + filename);
-        removeAndRestoreDataBundle(testData);
+        verifyOrPersistTestDataToDatastore();
+        // removeAndRestoreDataBundle(testData);
+    }
+
+    // verify if entities for testing already exist in datastore
+    private void verifyOrPersistTestDataToDatastore() {
+        log.info("Verifying test data existance in datastore...");
+
+        for (CourseAttributes course : testData.courses.values()) {
+            if (getCourse(course) == null) {
+                doPutCourse(course);
+                log.info("Course was added to datastore: " + course);
+            }
+        }
+
+        for (AccountAttributes account : testData.accounts.values()) {
+            if (getAccount(account) == null) {
+                doPutAccount(account);
+                log.info("Account was added to datastore: " + account);
+            }
+        }
+
+        for (FeedbackSessionAttributes feedbackSession : testData.feedbackSessions.values()) {
+            if (getFeedbackSession(feedbackSession) == null) {
+                doPutFeedbackSession(feedbackSession);
+                log.info("Feedback Session was added to datastore: " + feedbackSession);
+            }
+        }
+
+        for (InstructorAttributes instructor : testData.instructors.values()) {
+            if (getInstructor(instructor) == null) {
+                doPutInstructor(instructor);
+                log.info("Instructor was added to datastore: " + instructor);
+            }
+        }
+
+        int countStudents = 0;
+        for (StudentAttributes student : testData.students.values()) {
+            if (getStudent(student) == null) {
+                doPutStudent(student);
+                log.info("Student was added to datastore (" + countStudents + "): " + student);
+            }
+        }
+
+        int countFeedbackQuestions = 0;
+        int countFeedbackResponses = 0;
+        for (FeedbackQuestionAttributes question : testData.feedbackQuestions.values()) {
+            if (getFeedbackQuestion(question) == null) {
+                doPutFeedbackQuestion(question);
+                log.info("Feedback Question was added to datastore (" + countFeedbackQuestions + "): " + question);
+                countFeedbackQuestions++;
+            }
+            String feedbackQuestionRealId = getFeedbackQuestion(question).getId();
+            for (FeedbackResponseAttributes feedbackResponse : testData.feedbackResponses.values()) {
```

Comment:
Looks like you're looping through **all** feedback responses per question? Is that necessary?